nupic/research/monitor_mixin/temporal_memory_monitor_mixin.py

```diff
@@ -23,14 +23,18 @@
 Temporal Memory mixin that enables detailed monitoring of history.
 """
 
-from collections import defaultdict
+import numpy
```

Comment:
External imports are anything non-nupic related that we would have to add to `requirements.txt`. Built-in imports are libraries that come with Python (like `collections`).

So the order is:

1. Built-in imports
2. External imports
3. NuPIC imports (from other repositories)
4. Local imports (packages from this repository)